### PR TITLE
Sandbox example of User Tenant Service

### LINKED ISSUE #40: user-tenant-services-sandbox example
**Is your feature request related to a problem? Please describe.**
User-Tenant-Services sandbox example created ,so that A consumer/user can easily access the Api's through swagger Open API.

**Describe the solution you'd like**
“The OpenAPI Specification (OAS) defines a standard, language-agnostic interface to RESTful APIs which allows both humans and computers to discover and understand the capabilities of the service without access to source code, documentation, or through network traffic inspection.”

**Describe alternatives you've considered**

**Additional context**

### LINKED ISSUE #43: Permission Check
**Is your feature request related to a problem? Please describe.**
Code level permission added and use cases are listed below.

Eg : 1. If user has VIEW_ANY_USER permission then then is no need to compare  the tenantId with CurrentUser tenantId as user has any permission to view .
2.Similarly  If user does not have  VIEW_ANY_USER permission then it is require to compare  the tenantId with CurrentUser tenantId as user should not be able to see other tenant data.

3.If user has  UPDATE_TENANT permission then we check if the data is present against userId and TenantId ,
if yes we update the data else return the corresponding error,
Similarly if user has UPDATE_ANY_USER permission the there no need to check whether data is present against userId and TenantId  because user can update any user.
4. Similarly  more checks are added wherever required.

**Describe the solution you'd like**

**Conclusion**: Method level permission are 1st level check but these code level check are also required as explain above.

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by AdilSDE**
## Description
User-Tenant-Services sandbox example created ,so that A consumer/user can easily access the Api's through swagger Open API.

“The OpenAPI Specification (OAS) defines a standard, language-agnostic interface to RESTful APIs which allows both humans and computers to discover and understand the capabilities of the service without access to source code, documentation, or through network traffic inspection.”

Fixes #40
Fixes  #43 
Please delete options that are not relevant.

- [ ] New feature (non-breaking change which adds functionality)
- [ ] Intermediate change (work in progress)

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

## Checklist:

- [ ] Performed a self-review of my own code
- [ ] npm test passes on your machine
- [ ] New tests added or existing tests modified to cover all changes
- [ ] Code conforms with the style guide
- [ ] API Documentation in code was updated
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: services/user-tenant-service/src/main/java/com/sourcefuse/jarc/services/usertenantservice/controller/TenantController.java [hunk 1]
```diff
@@ -104,12 +144,25 @@ public ResponseEntity<String> updateTenantsById(
...
     tenantService.deleteById(id);
     return new ResponseEntity<>("Tenant DELETE success", HttpStatus.NO_CONTENT);
   }
 
   @GetMapping("/{id}/config")
+  @PreAuthorize(
+    "isAuthenticated() && hasAnyAuthority('" +
+    PermissionKeyConstants.VIEW_TENANT +
+    "'," +
+    "'" +
```
Comment (harshadk-sourcefuse) [inline services/user-tenant-service/src/main/java/com/sourcefuse/jarc/services/usertenantservice/controller/TenantController.java]:
```suggestion
"','" +
```

**Review by harshadk-sourcefuse — CHANGES_REQUESTED** [review]
check for similar changes as well in another files

File: services/user-tenant-service/src/test/java/com/sourcefuse/jarc/services/usertenantservice/unit/TenantUserControllerUnitTests.java [hunk 2]
```diff
@@ -474,15 +597,22 @@ void testFindByIdFound() {
...
+    CurrentUser currentUser=MockCurrentUserSession.getCurrentUser();
     // Act & Assert
     ResponseStatusException exception = Assertions.assertThrows(
       ResponseStatusException.class,
-      () -> tenantUserService.findById(userId)
+      () ->
+        tenantUserService.findById(
+          userId,
+          tenantId,
+                currentUser
```
Comment (mlakdawala-sf) [inline services/user-tenant-service/src/test/java/com/sourcefuse/jarc/services/usertenantservice/unit/TenantUserControllerUnitTests.java]: formatting issue

File: services/user-tenant-service/src/test/java/com/sourcefuse/jarc/services/usertenantservice/unit/TenantUserControllerUnitTests.java [hunk 3]
```diff
@@ -331,15 +316,52 @@ void testGetUserViewSuccess() {
...
+      )
+      .thenReturn(userViewsList);
+    CurrentUser currentUser=MockCurrentUserSession.getCurrentUser();
+    // Act
+    ResponseStatusException exception = Assertions.assertThrows(
+      ResponseStatusException.class,
+      () ->
+        tenantUserService.getUserView(
+          MockTenantUser.TENANT_ID,
+                currentUser
```
Comment (mlakdawala-sf) [inline services/user-tenant-service/src/test/java/com/sourcefuse/jarc/services/usertenantservice/unit/TenantUserControllerUnitTests.java]: formatting issue

File: services/user-tenant-service/src/main/java/com/sourcefuse/jarc/services/usertenantservice/controller/UserTenantController.java [hunk 4]
```diff
@@ -16,11 +19,25 @@
...
 public class UserTenantController {
 
   private final UserTenantService userTenantService;
 
   @GetMapping("{id}")
+  @PreAuthorize(
+    "isAuthenticated() && hasAnyAuthority('" +
+    PermissionKeyConstants.VIEW_ANY_USER +
+    "'," +
+    "'" +
```
Comment (mlakdawala-sf) [inline services/user-tenant-service/src/main/java/com/sourcefuse/jarc/services/usertenantservice/controller/UserTenantController.java]: can we make this in a single quote statement, and fix this in other places

File: services/user-tenant-service/src/main/java/com/sourcefuse/jarc/services/usertenantservice/controller/UserGroupController.java [hunk 5]
```diff
@@ -105,6 +132,11 @@ public ResponseEntity<List<UserGroup>> getAllUserGroupByGroupId(
   }
 
   @GetMapping("{id}/user-groups/count")
+  @PreAuthorize(
+    "isAuthenticated() && hasAnyAuthority('" +
```
Comment (mlakdawala-sf) [inline services/user-tenant-service/src/main/java/com/sourcefuse/jarc/services/usertenantservice/controller/UserGroupController.java]:
```suggestion
    "isAuthenticated() && hasAuthority('" +
```

Comment (mlakdawala-sf) [reply services/user-tenant-service/src/main/java/com/sourcefuse/jarc/services/usertenantservice/controller/UserGroupController.java]: Has any authority for multiple permission case, where the api can run through either of the permissions provided

File: services/user-tenant-service/src/main/java/com/sourcefuse/jarc/services/usertenantservice/controller/TenantUserController.java [hunk 6]
```diff
@@ -105,20 +143,43 @@ public ResponseEntity<CountResponse> userTenantCount(
...
+    UserView userView = tenantUserService.findById(userId, id, currentUser);
     return new ResponseEntity<>(userView, HttpStatus.OK);
   }
 
   @Transactional
   @PatchMapping("/{userId}")
+  @PreAuthorize(
+    "isAuthenticated() && hasAnyAuthority('" +
+    PermissionKeyConstants.UPDATE_ANY_USER +
+    "'," +
```
Comment (mlakdawala-sf) [inline services/user-tenant-service/src/main/java/com/sourcefuse/jarc/services/usertenantservice/controller/TenantUserController.java]: please fix it across the app

**Review by mlakdawala-sf — CHANGES_REQUESTED** [review]

File: sandbox/user-tenant-service-example/src/main/java/com/example/usertenantserviceexample/controller/PingController.java [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/ping")
+public class PingController {
+
+  @GetMapping
+  public ResponseEntity<String> ping() {
```
Comment (mlakdawala-sf) [inline sandbox/user-tenant-service-example/src/main/java/com/example/usertenantserviceexample/controller/PingController.java]: use health check package, check pom file of auth service

**Review by mlakdawala-sf — CHANGES_REQUESTED** [review]

**Review by mlakdawala-sf — APPROVED** [review]

**Review by harshadk-sourcefuse — APPROVED** [review]

**Pull request merged**